tests/kernel/interrupt/src/direct_isr.c

```diff
@@ -0,0 +1,139 @@
+/*
+ * Copyright (c) 2021 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <ztest.h>
+#include "interrupt_util.h"
+
+/*
+ * Other arch has already been tested in testcase of gen_isr_table,
+ * so we only test x86 and arch posix here.
+ * And x86_64 is also not support this kind of interrupt, we skip it.
+ */
+#if (defined(CONFIG_X86) && !defined(CONFIG_X86_64)) ||\
+	defined(CONFIG_ARCH_POSIX)
+
+#if defined(CONFIG_X86)
+
+#define TEST_DIRECT_IRQ_LINE_1	3
+#define TEST_DIRECT_IRQ_LINE_2	4
+
+/* the vector got would be different when enable code coverage */
+#if defined(CONFIG_COVERAGE)
+#define TRIGGER_DIRECT_IRQ_LINE_1	33
+#define TRIGGER_DIRECT_IRQ_LINE_2	34
+#else
+#define TRIGGER_DIRECT_IRQ_LINE_1	34
+#define TRIGGER_DIRECT_IRQ_LINE_2	33
+#endif
+
+#define TEST_DIRECT_IRQ_PRIO	0
+
+#elif defined(CONFIG_ARCH_POSIX)
+#define TEST_DIRECT_IRQ_LINE_1	5
+#define TEST_DIRECT_IRQ_LINE_2	6
+
+#define TRIGGER_DIRECT_IRQ_LINE_1	TEST_DIRECT_IRQ_LINE_1
+#define TRIGGER_DIRECT_IRQ_LINE_2	TEST_DIRECT_IRQ_LINE_2
+
+#define TEST_DIRECT_IRQ_PRIO	1
+
+#endif
+
+volatile uint32_t direct_int_executed[2];
+
+ISR_DIRECT_DECLARE(direct_isr1)
+{
+	direct_int_executed[0]++;
+
+	return 0;
```

Comment:
Should be there `return 1` too like in the macro below?

Reply:
Hi @maksimmasalski , it should return zero here. There return value decides if a swap should happen after direct ISR. You can check "void arch_isr_direct_footer(int swap)" for more details, thanks!